primary/src/block_waiter.rs

```diff
@@ -73,6 +76,24 @@ pub struct BatchMessage {
     pub transactions: Batch,
 }
 
+impl From<BatchMessage> for BatchMessageProto {
+    fn from(message: BatchMessage) -> Self {
+        BatchMessageProto {
+            id: Some(message.id.into()),
+            transactions: Some(BatchProto {
+                transaction: message
+                    .transactions
+                    .0
+                    .iter()
+                    .map(|transaction| TransactionProto {
+                        f_bytes: transaction.to_vec(),
+                    })
+                    .collect::<Vec<TransactionProto>>(),
+            }),
+        }
+    }
+}
+
```

Comment:
Yeah as was mentioned in another comment, if possible I think it would be great if we can centralize where these types are defined (eg in the `types` crate) and then include the from impls there.